pyprecag/processing.py

```diff
@@ -2537,3 +2401,222 @@ def ttest_analysis(points_geodataframe, points_crs, values_raster, out_folder,
+
+        if len(arg_check[-1]) == 0:
+            raise TypeError('Invalid Type: {} is a empty list'.format(arg_check[0]))
+
+    for arg_check in [('upper_percentage', upper_percentage),
+                      ('upper_probability', upper_probability),
+                      ('lower_percentage', lower_percentage),
+                      ('lower_probability', lower_probability)]:
+        if not isinstance(arg_check[1], int):
+            raise TypeError('{} must an integer'.format(arg_check[0]))
+
+    for arg_check in [('upper_percentage', upper_percentage),
+                      ('lower_percentage', lower_percentage)]:
+        if not isinstance(arg_check[1], int):
+            raise ValueError('{} must an integer between -50 to 50.'.format(arg_check[0]))
+
+    if output_tif is None or output_tif == '':
+        raise TypeError('Please specify an output filename')
+
+    if not os.path.exists(os.path.dirname(output_tif)):
+        raise IOError('Output directory {} does not exist'.format(os.path.dirname(output_tif)))
+
+    start_time = time.time()
+    temp_file_list = []
+    out_prefix, out_ext = os.path.splitext(os.path.basename(output_tif))
+
+    # Process upper ------------------------------------------------------------------------------
+    temp_file_list = [os.path.join(TEMPDIR, out_prefix + '_1_upperyears' + out_ext)]
+    upper_tif = persistor_all_years(upper_raster_files,
+                                    temp_file_list[-1],
+                                    greater_than=True,
+                                    target_percentage=upper_percentage)
+
+    upper_cutoff = (upper_probability / 100.00) * len(upper_raster_files)
+    with rasterio.open(upper_tif, 'r') as src:
+        kwargs = src.meta.copy()
+        band = src.read(masked=True)
+
+        # Greater than test = retuns boolean which can be converted to 0,1
+        data_u = np.where(~band.mask, (band >= upper_cutoff).astype(np.int16), -9999)
+
+    if config.get_debug_mode():
+        temp_file_list += [os.path.join(TEMPDIR, out_prefix + '_2gt_upperprob' + out_ext)]
+        with rasterio.open(temp_file_list[-1], 'w', **kwargs) as dst:
+            dst.write(data_u)
+
+    del band
+
+    # Process lower ------------------------------------------------------------------------------
+    temp_file_list += [os.path.join(TEMPDIR, out_prefix + '_1_loweryears' + out_ext)]
+    lower_tif = persistor_all_years(lower_raster_files,
+                                    temp_file_list[-1],
+                                    greater_than=False,
+                                    target_percentage=lower_percentage)
+
+    lower_cutoff = (lower_probability / 100.00) * len(lower_raster_files)
+    with rasterio.open(lower_tif, 'r') as src:
+        band = src.read(masked=True)
+
+        # Greater than test = retuns boolean which can be converted to 0,1
+        data_l = np.where(~band.mask, np.negative((band >= lower_cutoff).astype(np.int16)), -9999)
```

Comment:
Here, since you are using the negative, don't change ">=" to ">"

Reply:
`(band >= lower_cutoff)` returns booleans (0,1) we need -1, 0 so convert to integer and use np.negative and maintain the nodata values. Previously this was done via the GreaterThanEqual tool then a reclass ( 1 1 -1).